data-prepper-plugins/csv-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/csv/CsvProcessor.java

```diff
@@ -210,7 +213,11 @@ private List<String> parseHeaderFromEventSourceKey(final Event event, final CsvM
     private void putDataInEvent(final Event event, final List<String> header, final List<String> data) {
         int providedHeaderColIdx = 0;
         for (; providedHeaderColIdx < header.size() && providedHeaderColIdx < data.size(); providedHeaderColIdx++) {
-            event.put(header.get(providedHeaderColIdx), data.get(providedHeaderColIdx));
+            String key = header.get(providedHeaderColIdx);
+            if (normalizeKeys) {
+                key = JacksonEvent.replaceInvalidKeyChars(key);
+            }
+            event.put(key, data.get(providedHeaderColIdx));
```

Comment:
This could be:

```
event.put(key, data.get(providedHeaderColIdx, true);
```

Where the last boolean is a property to replace invalid keys.